meson.build

```diff
@@ -0,0 +1,484 @@
+endif
+if cc.has_header_symbol('unistd.h', 'eaccess', prefix: '#define _GNU_SOURCE')
+    configh_data.set('HAVE_EACCESS', 1)
+endif
+if cc.has_header_symbol('unistd.h', 'euidaccess', prefix: '#define _GNU_SOURCE')
+    configh_data.set('HAVE_EUIDACCESS', 1)
+endif
+if cc.has_header_symbol('sys/mman.h', 'mmap')
+    configh_data.set('HAVE_MMAP', 1)
+endif
+if cc.has_header_symbol('stdlib.h', 'secure_getenv', prefix: '#define _GNU_SOURCE')
+    configh_data.set('HAVE_SECURE_GETENV', 1)
+elif cc.has_header_symbol('stdlib.h', '__secure_getenv', prefix: '#define _GNU_SOURCE')
+    configh_data.set('HAVE___SECURE_GETENV', 1)
+else
+    message('C library does not support secure_getenv, using getenv instead')
+endif
+configure_file(output: 'config.h', configuration: configh_data)
+add_project_arguments('-include', 'config.h', language: 'c')
+
+
+# Supports -Wl,--version-script?
+have_version_script = cc.links(
+    'int main(){}',
+    link_args: '-Wl,--version-script=' + join_paths(meson.source_root(), 'xkbcommon.map'),
+    name: '-Wl,--version-script',
+)
+
+
+# libxkbcommon.
+# Note: we use some yacc extensions, which work with either GNU bison
+# (preferred) or byacc. Other yacc's may or may not work.
+yacc = find_program('bison', 'byacc')
+yacc_gen = generator(
+    yacc,
+    output: ['@BASENAME@.c', '@BASENAME@.h'],
+    arguments: ['@INPUT@', '--defines=@OUTPUT1@', '--output=@OUTPUT0@', '-p _xkbcommon_'],
+)
+libxkbcommon_internal = static_library(
+    'xkbcommon-internal',
+    'src/compose/parser.c',
+    'src/compose/parser.h',
+    'src/compose/paths.c',
+    'src/compose/paths.h',
+    'src/compose/state.c',
+    'src/compose/table.c',
+    'src/compose/table.h',
+    'src/xkbcomp/action.c',
+    'src/xkbcomp/action.h',
+    'src/xkbcomp/ast.h',
+    'src/xkbcomp/ast-build.c',
+    'src/xkbcomp/ast-build.h',
+    'src/xkbcomp/compat.c',
+    'src/xkbcomp/expr.c',
+    'src/xkbcomp/expr.h',
+    'src/xkbcomp/include.c',
+    'src/xkbcomp/include.h',
+    'src/xkbcomp/keycodes.c',
+    'src/xkbcomp/keymap.c',
+    'src/xkbcomp/keymap-dump.c',
+    'src/xkbcomp/keywords.c',
+    yacc_gen.process('src/xkbcomp/parser.y'),
+    'src/xkbcomp/parser-priv.h',
+    'src/xkbcomp/rules.c',
+    'src/xkbcomp/rules.h',
+    'src/xkbcomp/scanner.c',
+    'src/xkbcomp/symbols.c',
+    'src/xkbcomp/types.c',
+    'src/xkbcomp/vmod.c',
+    'src/xkbcomp/vmod.h',
+    'src/xkbcomp/xkbcomp.c',
+    'src/xkbcomp/xkbcomp-priv.h',
+    'src/atom.c',
+    'src/atom.h',
+    'src/context.c',
+    'src/context.h',
+    'src/context-priv.c',
+    'src/darray.h',
+    'src/keysym.c',
+    'src/keysym.h',
+    'src/keysym-utf.c',
+    'src/ks_tables.h',
+    'src/keymap.c',
+    'src/keymap.h',
+    'src/keymap-priv.c',
+    'src/scanner-utils.h',
+    'src/state.c',
+    'src/text.c',
+    'src/text.h',
+    'src/utf8.c',
+    'src/utf8.h',
+    'src/utils.c',
+    'src/utils.h',
+    include_directories: include_directories('src'),
```

Comment:
Yes, from my tests it works as it's supposed to. I assume the source file's own directory comes before the `include_directories` in the search path.